buildstockbatch/cloud/docker_base.py

```diff
@@ -461,3 +462,48 @@ def run_simulations(cls, cfg, job_id, jobs_d, sim_dir, fs, output_path):
                 shutil.rmtree(item)
             elif os.path.isfile(item):
                 os.remove(item)
+
+    def log_summary(self):
+        """
+        Log a summary of how many simulations succeeded, failed, or ended with other statuses.
+
+        Uses the `completed_status` column of the files in results_csvs/results_*.csv.gz.
+        """
+        fs = self.get_fs()
+        # Summary of simulation statuses across all upgrades
+        status_summary = {}
+        all_statuses = set()
+
+        results_csv_dir = f"{self.results_dir}/results_csvs/"
+        try:
+            results_files = fs.ls(results_csv_dir)
+        except FileNotFoundError:
+            logger.info(f"No results CSV files found at {results_csv_dir}")
+            return
+
+        for result in results_files:
+            upgrade_id = result.split(".")[0][-2:]
+            with fs.open(result) as f:
+                with gzip.open(f) as gf:
+                    df = pd.read_csv(gf, usecols=["completed_status"])
+            # Dict mapping from status (e.g. "Success") to count
+            statuses = df.groupby("completed_status").size().to_dict()
+            status_summary[upgrade_id] = statuses
+            all_statuses.update(statuses.keys())
+
+        # Always include these statuses and show them first
+        always_use = ["Success", "Fail"]
+        all_statuses = always_use + list(all_statuses - set(always_use))
+        s = "Final status of all simulations:"
+        for upgrade, counts in status_summary.items():
+            if upgrade == "00":
+                s += "\nBaseline     "
+            else:
+                s += f"\nUpgrade {upgrade}   "
+            for status in all_statuses:
+                s += f"{status}: {counts.get(status, 0):<7d}  "
+
+        for upgrade in postprocessing.get_upgrade_list(self.cfg):
+            if f"{upgrade:02d}" not in status_summary:
+                s += f"\nNo results found for Upgrade {upgrade}"
+        logger.info(s)
```

Comment:
What do you think of also including:

* The sum for each status (e.g., total successes and total failures)

* The total of all statuses counted, and a warning (of sorts) if it's less than the total number of sims?

Reply:
Sums: Sure, I can add that.

Comparing to the total: This is possible, but kind of annoying to do at this point in the job. To find the total number of simulations that _should_ exist, I think we have to pull buildstock.csv back out of assets.tar.gz on the cloud and re-count how many rows it has. (We already have this info locally if running the full pipeline, but may not if running postprocessing only.) Thoughts on how important this is? Or a simpler way to find the intended total?

Reply:
If the total isn't readily available, I think it's fine to leave it out. We can easily pull that from elsewhere if needed.